modules/core/src/main/java/org/apache/ignite/internal/direct/stream/v2/DirectByteBufferStreamImplV2.java

```diff
@@ -584,7 +587,17 @@ public DirectByteBufferStreamImplV2(MessageFactory msgFactory) {
 
     /** {@inheritDoc} */
     @Override public void writeString(String val) {
-        writeByteArray(val != null ? val.getBytes() : null);
+        if (val != null) {
+            if (byteArr == null)
+                byteArr = val.getBytes();
+
+            writeByteArray(byteArr);
+
+            if (lastFinished)
+                byteArr = null;
+        }
+        else
+            writeByteArray(null);
```

Comment:
can we perhaps declare byteArr a local variable, and only assign it to curStrBackingArr if not lastFinished? This way we will not touch this variable at all when string fits one frame.

Reply:
Actually, we need to reset it to null if lastFinished. Need to figure out the logic, maybe it's too cumbersome and confusing to implement.

Reply:
Disregard this, we are going to read curStrBackingArr value either way. I think we may skip this change entirely, just rename var / mark volatile.